x-pack/plugins/cases/public/components/case_view/components/custom_fields.test.tsx

```diff
@@ -160,130 +162,90 @@ describe('Case View Page files tab', () => {
-        {
-          type: CustomFieldTypes.TEXT,
-          key: 'test_key_1',
-          value: customFieldsConfigurationMock[0].defaultValue,
-        },
-        {
-          type: CustomFieldTypes.TOGGLE,
-          key: 'test_key_2',
-          value: false,
-        },
-      ]);
+      expect(onSubmit).toBeCalledWith({
+        type: CustomFieldTypes.TOGGLE,
+        key: 'test_key_2',
+        value: true,
+      });
     });
   });
 
-  it('does not overwrite existing text values with a configured defaultValue', async () => {
+  it('replaces existing text field correctly', async () => {
     appMockRender.render(
       <CustomFields
         isLoading={false}
-        customFields={[
-          { key: customFieldsMock[0].key, type: CustomFieldTypes.TEXT, value: 'existing value' },
-          { ...customFieldsMock[1] },
-        ]}
-        customFieldsConfiguration={[
-          customFieldsConfigurationMock[0],
-          customFieldsConfigurationMock[1],
-        ]}
+        customFields={customFieldsMock}
+        customFieldsConfiguration={customFieldsConfigurationMock}
         onSubmit={onSubmit}
       />
     );
 
-    userEvent.click((await screen.findAllByRole('switch'))[0]);
-
-    await waitFor(() => {
-      expect(onSubmit).toBeCalledWith([
-        {
-          type: CustomFieldTypes.TEXT,
-          key: 'test_key_1',
-          value: 'existing value',
-        },
-        {
-          type: CustomFieldTypes.TOGGLE,
-          key: 'test_key_2',
-          value: false,
-        },
-      ]);
-    });
-  });
+    userEvent.click(
+      await screen.findByTestId(`case-text-custom-field-edit-button-${customFieldsMock[0].key}`)
+    );
 
-  it('removes extra custom fields', async () => {
-    appMockRender.render(
-      <CustomFields
-        isLoading={false}
-        customFields={customFieldsMock}
-        customFieldsConfiguration={[
-          {
-            type: CustomFieldTypes.TOGGLE,
-            key: 'test_key_2',
-            label: 'My test label 2',
-            required: false,
-          },
-        ]}
-        onSubmit={onSubmit}
-      />
+    userEvent.paste(
+      await screen.findByTestId('case-text-custom-field-form-field-test_key_1'),
+      '!!!'
     );
 
-    userEvent.click(await screen.findByRole('switch'));
+    userEvent.click(await screen.findByTestId('case-text-custom-field-submit-button-test_key_1'));
 
     await waitFor(() => {
-      expect(onSubmit).toBeCalledWith([
-        { type: CustomFieldTypes.TOGGLE, key: 'test_key_2', value: false },
-      ]);
+      expect(onSubmit).toBeCalledWith({
+        ...customFieldsMock[0],
+        value: 'My text test value 1!!!',
+      });
     });
   });
 
-  it('updates an existing field correctly', async () => {
+  it('replaces new text field correctly', async () => {
     appMockRender.render(
       <CustomFields
         isLoading={false}
-        customFields={customFieldsMock}
+        customFields={[]}
```

Comment:
This means the value will be prepopulated with the default.

Could you please also look for the `This field is populated with the default value.` message in this text?